### PR TITLE
Update qrcode

### PR DESCRIPTION
**Pull request by ankitdas13**
Api Doc: https://razorpay.com/docs/api/qr-codes/gst/
Ref: https://github.com/razorpay/razorpay-php/blob/master/documents/qrcode.md

### CODE REVIEW

File: resources/qr_code.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// Create creates a new qrcode for the given data.
+func (q *QrCode) Create(data map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	return q.Request.Post(constants.QRCODE_URL, data, extraHeaders)
+}
+
+//Fetch fetches the qrcode entity having the given QrCodeID.
+func (q *QrCode) FetchPayments(QrCodeID string, queryParams map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("%s/%s/payments", constants.QRCODE_URL, QrCodeID)
```
Comment (Madeyedexter) [inline resources/qr_code.go:22]: While building the url, the QrCodeId should be url escaped as the input Id can be passed with a `/` and can indicate a different url.


Please check other urls as well that are being built via `fmt.Sprintf`

File: testdata/fake_qr_code.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "payments_amount_received": 0,
+    "payments_count_received": 0,
+    "notes": {
+      "purpose": "Test UPI QR code notes"
+    },
+    "customer_id": "cust_HKsR5se84c5LTO",
+    "close_by": 1681615838,
+    "closed_at": null,
+    "close_reason": null
+  }
```
Comment (Madeyedexter) [inline testdata/fake_qr_code.json:22]: Ensure an EOL char in all json files

Comment (Madeyedexter) [inline testdata/fake_qr_code.json:22]: Ensure an EOL

Comment (ankitdas13) [reply testdata/fake_qr_code.json]: done in all qr json files

File: resources/qr_code.go (hunk 1 above)
Comment (ankitdas13) [reply resources/qr_code.go]: added escape in all get methods api's

File: resources/qr_code.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+//Close close a QrCode for the given QrCodeID.
+func (q *QrCode) Close(QrCodeID string, data map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("%s/%s/close", constants.QRCODE_URL, url.PathEscape(QrCodeID))
+	return q.Request.Post(url, data, extraHeaders)
+}
+
+
+
+
```
Comment (Madeyedexter) [inline resources/qr_code.go:46]: Remove extra lines

File: testdata/fake_qr_code.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "payments_amount_received": 0,
+    "payments_count_received": 0,
+    "notes": {
+      "purpose": "Test UPI QR code notes"
+    },
+    "customer_id": "cust_HKsR5se84c5LTO",
+    "close_by": 1681615838,
+    "closed_at": null,
+    "close_reason": null
+}
```
Comment (Madeyedexter) [inline testdata/fake_qr_code.json:22]: Please add EOL character at the end of each JSON file

**Review by Madeyedexter — APPROVED** [review]

**Pull request merged**